test/test_nestedtensor.py

```diff
@@ -1230,6 +1230,25 @@ def test_device_checks(self, device):
         is_cuda = "cuda" in str(device)
         self.assertEqual(nt.is_cuda, is_cuda)
 
+    @skipIfTorchDynamo("Not a suitable test for TorchDynamo")
+    def test_share_memory(self, device):
+        a = torch.randn(3, 4, device=device)
+        b = torch.randn(5, 4, device=device)
+        nt = torch.nested.nested_tensor([a, b], layout=torch.jagged)
+
+        # Guard CUDA tensors
+        if device == "cuda":
```

Comment:
```suggestion
        if "cuda" in device:
```

this is more accurate because in practice `device` will be something like `"cuda:0"`